### PR TITLE
Migrate .hgignore to .gitignore

### LINKED ISSUE #6: Improve .gitignore file
The `.gitignore` file should be as well documented and should cover as many cases as the `.hgignore` file, now that many projects are using Git as the VCS.

### LINKED ISSUE #8: Migrate .hgignore to .gitignore

### PR DESCRIPTION
**Pull request by anita-chouhan**
Fixes #8, Fixes #6

### CODE REVIEW

**Comment by rungta** [comment]
The `.hgignore` file should be removed completely.

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,36 @@
-assets/node_modules
-web/build
-assets/package-lock.json
+# OS Cruft
+.DS_Store
+
+# Package Managers
+vendor
+node_modules/
```
Comment (rungta) [inline .gitignore:6]: Why does one of these have a trialing slash and the other does not?

**Review by rungta — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (anita-chouhan) [reply .gitignore]: Fixed. Left by mistake.

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,36 @@
...
+# OS Cruft
+.DS_Store
+
+# Package Managers
+vendor
+node_modules
+
+# Craft CMS
+config/license.key
+web/cpresources/
```
Comment (rungta) [inline .gitignore:10]: Trailing slash. ☹️

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,36 @@
...
+/htpasswd
+assets/.env
+web/.htaccess
+.tm_properties
+*.sublime-project
+*.sublime-workspace
+*.code-workspace
+syncCraft.*.cfg
+syncCraft.sh
+.idea/
```
Comment (rungta) [inline .gitignore:25]: Trialing slash.

File: .gitignore [hunk 4]
```diff
@@ -1,3 +1,36 @@
...
+.vscode/
```
Comment (rungta) [inline .gitignore:26]: Trailing slash.

**Review by rungta — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 2 above)
Comment (anita-chouhan) [reply .gitignore:10]: .gitignore has a different [usage](https://github.com/git/git/blob/1a58bad0144dedd59eeef29b7ba7791aed19106f/Documentation/gitignore.txt#L92-L113) for with and without trailing slash patterns so not modified it intentionally.

Comment (anita-chouhan) [reply .gitignore:10]: > Let’s fix that inconsistency now.

Added trailing slashes to all the folder names.

**Review by rungta — APPROVED** [review]

**Pull request merged**